src/main/scala/ChannelMapper.scala

```diff
@@ -10,10 +11,10 @@ import cde.Parameters
  * channel are contiguous.
  */
 class ChannelAddressMapper(n: Int)(implicit p: Parameters) extends Module {
-  val io = new Bundle {
+  val io = IO(new Bundle {
     val in  = Vec(n, new ClientUncachedTileLinkIO).flip
     val out = Vec(n, new ClientUncachedTileLinkIO)
```

Comment:
ClientUncachedTileLinkIO is defined in the Chisel emulation layer, so all the ports should have OUTPUT or INPUT (which presumably calls Output and Input?). This is actually something I'm not sure I understand- what are you meant to do when your port is a bundle (i.e. something that doesn't just have one direction? Should it be wrapped in Output()?